common/network/CMakeLists.txt

```diff
@@ -1,6 +1,7 @@
 include_directories(${CMAKE_SOURCE_DIR}/common)
 
 add_library(network STATIC
+  UnixSocket.cxx
```

Comment:
And you can include `UnixSocket.cxx` condtionally. Avoids having a file that is completely `#ifdef`ed out.